src/interfaces/cartItem.d.ts

```diff
@@ -1,7 +1,7 @@
 import { IDesign } from "@/interfaces/design";
 import { PriceBar } from "@/store/reducer/priceBarSlice";
 
-interface ICartItem {
+export interface ICartItem {
```

Comment:
Is it really necessary to write export? From my understanding it may not, but maybe I'm wrong, it would be great if u check it again, thank you.

Reply:
You need to export the interfaces before you import them in your component.

Reply:
> Is it really necessary to write export? From my understanding it may not, but maybe I'm wrong, it would be great if u check it again, thank you.
You need to export the interfaces before you import them in your component.

Reply:
have you tried remove export and see whether there are error in the program? :)

Reply:
> have you tried remove export and see whether there are errors in the program? :)

Currently there are no error, however If you do not add "export" before the interface, it will become a global variables, which would be very easy to cause variable naming conflict in the future.

Reply:
Thanks for your explaination!

Reply:
> 

No error is not the only way to identify a good code or a bad one. It should be the bottom line of a pass mark.

For this case, I think use a plain ts file is even better than d.ts.

Reply:
> > 
> 
> No error is not the only way to identify a good code or a bad one. It should be the bottom line of a pass mark.
> 
> For this case, I think use a plain ts file is even better than d.ts.

Understood!